lib/eco/treemanager.py

```diff
@@ -241,8 +241,11 @@ def find_next_visible(self, node):
     def find_previous_visible(self, node, cross_lang=True):
         """Return the previous visible node in the parse tree."""
         if self.is_visible(node):
-            node = node.previous_terminal() # XXX check for multiterm
-        while not self.is_visible(node):
+            if type(node.symbol) is MagicTerminal and cross_lang:
+                node = node.symbol.ast.children[-1]
+            else:
+                node = node.previous_terminal() # XXX check for multiterm
```

Comment:
Is the `XXX` still relevant?

Reply:
Actually, it is not. Fixed in latest commit.